src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,129 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import * as commentsService from '../api/comments';
+import cn from 'classnames';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  addNewComment: (comment: Comment) => void;
+  setIsError: (isError: boolean) => void;
+  setIsAddingComment: Dispatch<SetStateAction<boolean>>;
+};
+
+export const NewCommentForm: React.FC<Props> = props => {
+  const { postId, addNewComment, setIsError, setIsAddingComment } = props;
+  const [name, setName] = useState('');
+  const [isNameError, setIsNameError] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+
+  const [commentText, setCommentText] = useState('');
+  const [isCommentError, setIsCommentError] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const reset = () => {
+    setIsNameError(false);
+    setIsEmailError(false);
+    setIsCommentError(false);
+    setName('');
+    setEmail('');
+    setCommentText('');
+    setIsLoading(false);
+  };
+
+  const validateform = () => {
```

Comment:
Typo in function name: `validateform` should be `validateForm` to follow camelCase naming conventions.